Add Access.values/0

xxdavid opened the pull request:
Hi, I am adding `Access.values/0` as discussed [here](https://groups.google.com/g/elixir-lang-core/c/z3dK1TEhwsg), and also `Access.keys/0` for parity with keys. If you think `Access.keys/0` is not useful and should not be added, I can remove it from the PR.

I am not adding any explicit unit tests as I find the coverage by doctests enough, but I can write some if you wish so.

Thank you for reviewing the changes!

xxdavid commented on lib/elixir/lib/access.ex:
```diff
@@ -999,6 +999,124 @@ defmodule Access do
+      iex> get_in([1, 2, 3], [Access.values()])
+      ** (RuntimeError) Access.values/0 expected a map, got: [1, 2, 3]
+  """
+  @spec values() :: Access.access_fun(data :: map(), current_value :: list())
```
Added in f7705f45. Thanks for noticing!

sabiwara reviewed (approved):
LGTM, thanks @xxdavid ! 💜

josevalim commented:
Let's not add `keys` for now but I'd suggest to make it so `values` works both with keyword lists and maps. WDYT?

xxdavid commented:
I added the support for keyword lists in the last commit. I was thinking about what to allow as a keyword list and I've come to a pragmatic decision to check whether the _first element_ of a list is a 2-tuple and the first element of that tuple is an atom. If  not, we'll raise with `Access.values/0 expected a map or a keyword list, got: …`. It may fail with `FunctionClauseError` if given for example `[{:a, 1}, {:b, 2, :haha}]` but I think that's acceptable as for example `Keyword.filter/1` would fail as well.

I've also added unit tests as it got a bit more complex. And I've also found that some doctests for `values/0` were relying on a particular or of keys in a map, which I fixed also in the last commit.

sabiwara commented on lib/elixir/lib/access.ex:
```diff
@@ -999,6 +999,93 @@ defmodule Access do
+  end
+
+  defp values_keyword(:get, data, next) do
+    Enum.map(data, fn {_key, value} -> next.(value) end)
+  end
+
+  defp values_keyword(:get_and_update, data, next) do
+    {reverse_gets, reverse_updated_data} =
+      Enum.reduce(data, {[], []}, fn {key, value}, {gets, data_acc} ->
```
Perhaps we should still have guards to be sure it's a keyword?
```suggestion
    Enum.map(data, fn {key, value} when is_atom(key) -> next.(value) end)
  end

  defp values_keyword(:get_and_update, data, next) do
    {reverse_gets, reverse_updated_data} =
      Enum.reduce(data, {[], []}, fn {key, value}, {gets, data_acc} when is_atom(key) ->
```

xxdavid replied:
I looked at `Keyword.filter/2` and `Keyword.values/1` and they don't seem to be that strict. But why not. 🙂 Added in 8b4aa0f7.

josevalim commented:
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
